### PR TITLE
[FIX] stock_landed_cost: efficient validate landed cost

### PR DESCRIPTION
**Pull request by ryv-odoo**
Update the standard_price outside the loop to avoid recompute
` quantity_svl` at each iteration. `quantity_svl` is not modify in the
process and the `standard_price` is not read in process.

 `NewPrice = OldPrice + (cost_to_add_x1) / quantity_svl + (cost_to_add_x2) / quantity_svl`
 Becomes
`NewPrice = OldPrice + (cost_to_add_x1 + cost_to_add_x2) / quantity_svl`
which is equals.

opw-2374405

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Review by nim-odoo — APPROVED** [review]
It looks fine. From a functional perspective, the order in which we create the AML should not be relevant in the standard price computation. Hence, it is logical to be able to take the standard price computation out of the AML creation loop.

This is verified by the fact that we don't use the `standard_price` in the AML creation: https://github.com/odoo/odoo/blob/893a3e63101fcbaad34b915c84cbf0f148ad8cc8/addons/stock_landed_costs/models/stock_landed_cost.py#L380-L465

File: addons/stock_landed_costs/models/stock_landed_cost.py [hunk 1]
```diff
@@ -168,6 +170,12 @@ def button_validate(self):
                     qty_out = line.move_id.product_qty
                 move_vals['line_ids'] += line._create_accounting_entries(move, qty_out)
 
+            # batch standard price computation avoid recompute quantity_svl at each iteration
+            products = self.env['product.product'].browse(p.id for p in cost_to_add_byproduct.keys())
+            for product in products:  # iterate on recordset to prefetch efficiently quantity_svl
+                if not float_is_zero(product.quantity_svl, precision_rounding=product.uom_id.rounding):
+                    product.with_context(force_company=self.company_id.id).sudo().standard_price += cost_to_add_byproduct[product] / product.quantity_svl
```
Comment (amoyaux) [inline addons/stock_landed_costs/models/stock_landed_cost.py:177]: cost.company_id

**Comment by amoyaux** [comment]
robodoo r+

**Comment by amoyaux** [comment]
robodoo rebase-ff

**Comment by amoyaux** [comment]
robodoo r+

**Pull request closed without merging**